### PR TITLE
BaSP-TG-115: Implement firebase app

### PR DESCRIPTION
**Pull request by renzopimentel**
- Install Firebase library
- Set .env
- Modify routes and create auth and private routes for the user roles
- Create Auth components for login and sign up
- Add redux authentications (actions, constants, reducer and thunks)

**For QA:**

- Checkout master
- Update master `git fetch` && `git pull origin master`
- Checkout to this branch `git checkout feature/TG-115/implement-firebase-app`
- Install the dependencies `npm i`
- Run in console `npm start`
- See the description in the PR of the ticket TG-115 to follow the test steps

### CODE REVIEW

File: src/routes/index.js [hunk 1]
```diff
@@ -35,28 +33,23 @@ function Layout() {
...
-          <Route exact path="/projects" component={Projects} />
-          <Route path="/projects/form" component={ProjectsForm} />
-          <Route path="/projects/:id" component={ProjectsForm} />
-          <Route exact path="/timesheets" component={TimeSheets} />
-          <Route path="/timesheets/form" component={TimeSheetsForm} />
-          <Route path="/timesheets/:id" component={TimeSheetsForm} />
-          <Route exact path="/tasks" component={Tasks} />
-          <Route path="/tasks/form" component={TasksForm} />
-          <Route path="/tasks/:id" component={TasksForm} />
+          <PrivateRoute path="/employees" role="EMPLOYEE" component={EmployeeRoutes} />
```
Comment (CristianBucci) [inline src/routes/index.js:36]: This should be exact path

File: src/redux/auth/actions.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  FIREBASE_LOGOUT_PENDING,
+  FIREBASE_LOGOUT_SUCCESS,
+  FIREBASE_LOGOUT_ERROR,
+  FIREBASE_SET_AUTH
+} from './constants';
+
+export const firebaseLoginPending = (data) => {
+  return {
+    type: FIREBASE_LOGIN_PENDING,
+    payload: data
```
Comment (CristianBucci) [inline src/redux/auth/actions.js:17]: The payload is not necessary here!

File: src/redux/auth/actions.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  return {
+    type: FIREBASE_LOGIN_ERROR,
+    payload: data
+  };
+};
+
+export const firebaseSignUpPending = (data) => {
+  return {
+    type: FIREBASE_SIGN_UP_PENDING,
+    payload: data
```
Comment (CristianBucci) [inline src/redux/auth/actions.js:38]: Same here!

File: src/redux/auth/actions.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  return {
+    type: FIREBASE_SIGN_UP_ERROR,
+    payload: data
+  };
+};
+
+export const firebaseLogoutPending = (data) => {
+  return {
+    type: FIREBASE_LOGOUT_PENDING,
+    payload: data
```
Comment (CristianBucci) [inline src/redux/auth/actions.js:59]: Same as above!

**Review by Ansitox — APPROVED** [review]

File: src/redux/auth/actions.js (hunk 2 above)
Comment (MaxiG-dev) [reply src/redux/auth/actions.js:17]: Done!

File: src/redux/auth/actions.js (hunk 3 above)
Comment (MaxiG-dev) [reply src/redux/auth/actions.js:38]: Done!

File: src/redux/auth/actions.js (hunk 4 above)
Comment (MaxiG-dev) [reply src/redux/auth/actions.js:59]: Done!

**Review by MaxiG-dev — APPROVED** [review]
Looks great to me!

**Review by CristianBucci — APPROVED** [review]
Everything looks fine!

File: src/Components/Auth/Login/index.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        <div>
+          <img src="/assets/images/spinner.gif" alt="spinner" />
+        </div>
+      ) : (
+        role &&
+        email && (
+          <>
+            <h1>
+              {' '}
+              Bienvenido {role} {email}{' '}
```
Comment (matescarabino) [inline src/Components/Auth/Login/index.js:35]: Remember to always write in English even if we are going to change it later. Please do not change it now...

**Review by matescarabino — APPROVED** [review]
LGTM

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

Implement firebase

We wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**